torch/distributions/multinomial.py

```diff
@@ -0,0 +1,87 @@
+import torch
+from torch.distributions.distribution import Distribution
+from torch.distributions import Categorical
+from numbers import Number
+from torch.distributions import constraints
+from torch.distributions.utils import log_sum_exp
+
+
+class Multinomial(Distribution):
+    r"""
+    Creates a Multinomial distribution parameterized by `probs` and `total_count`.
+
+    If `probs` is 1D with length-`K`, each element is the relative probability
+    of sampling the class at that index.
+
+    If `probs` is 2D, it is treated as a batch of probability vectors.
+
+    Example::
+
+        >>> m = Multinomial(10, torch.Tensor([ 1, 1, 1, 1]))
+        >>> m.sample()  # equal probability of 0, 1, 2, 3
+         2
+         2
+         3
+         3
+        [torch.FloatTensor of size 4]
+
+
+    Args:
+        total_count (Tensor or Variable): number of trials
+        probs (Tensor or Variable): event probabilities
+    """
+
+    params = {'total_count': constraints.nonnegative_integer, 'probs': constraints.simplex}
+
+    def __init__(self, total_count, probs=None, logits=None):
+        self.total_count = total_count if isinstance(total_count, Number) else total_count.data[0]
```

Comment:
I'm surprised this works; I would think `total_count` must be an integer (not a tensor), and doesn't really have an entry in `.params`. One option would be to default to `total_count=1` so that constraint tests pass. This also makes sense for users who simply want to `Multinomial(probs).log_prob(sample)` since `total_count` is not needed.